juju/juju
# ec2: add eu-west-3, regenerate instance types

## Description of change

Add the eu-west-3 region, and update/regenerate instance types from the latest data. Update the EBS storage provider to cater for M5/C5 instance types using NVMe.

## QA steps

1. juju bootstrap aws/eu-west-3
2. juju add-model foo aws/eu-west-3
3. juju deploy postgresql --storage pgdata=10G
4. juju add-model bar aws/us-west-2
5. juju deploy postgresql --constraints instance-type=c5.large --storage pgdata=10G
6. juju deploy postgresql --constraints instance-type=m5.large --storage pgdata=10G pg2

(verify all postgresql applications come up successfully, with EBS storage attached; confirm expected instance types and regions via the AWS console)

## Documentation changes

None.

## Bug reference

This partially addresses https://bugs.launchpad.net/juju/+bug/1730809:
only partially, because the latest pricing data from AWS does not consistently
provide data for C5 instances in all regions, even when support is there.
